libbeat/beat/beat.go

```diff
@@ -64,7 +64,7 @@ type Beat struct {
 
 	BeatConfig *common.Config // The beat's own configuration section
 
-	Fields []byte // Data from fields.yml
+	//Fields []byte // Data from fields.yml
```

Comment:
Why remove this? The libbeat/beat package is supposed to be the (most...) stable API beats devs are facing/interacting with. We should try to reduce the change of breakages... Maybe we've been better off if Fields would have been an interface or not been added at all to the Beats structure.

Was Fields even used?

Reply:
Afaics `Fields` was only used for creating the template, so it is not necessary any more with those changes.

Reply:
In case you consider removing this a breaking change, I can leave it. It wouldn't be used any more though, so to avoid confusion and errors, I prefer removing it.

Reply:
I did a quick look, I don't see it used outside of creating the template.
So it introduce a breaking change on the beat struct, we should have defined an interface for this to limit the risk.

Lets say that I've used Fields in my code how I move it to the new workflow?
Or How do I have access to the same data?